AK/StringView.cpp

```diff
@@ -95,7 +95,7 @@ Vector<StringView> StringView::split_view(const StringView& separator, bool keep
         view = view.substring_view_starting_after_substring(part_with_separator);
         maybe_separator_index = view.find_first_of(separator);
     }
-    if (keep_empty || !view.is_empty())
+    if (!view.is_empty())
```

Comment:
Does any of the other split functions have this same problem?
* `String::split()`
* `String::split_view()`
Also please add a test :)

Reply:
Thinking a little about this, I am a little lost on whether this is actually a correct behaviour, should `"axbx".split_view('x', true)` be `"a", "b"` or `"a", "b", ""`?

I can think of reasons of both being correct, so I instinctively reach for keeping it as-is...

Reply:
The idea with `keep_empty` is that no empty strings will appear in the output.

Reply:
Then I'll revert my change and just add some tests.